test/testfunctions.cpp

```diff
@@ -1922,6 +1922,12 @@ class TestFunctions : public TestFixture {
               "}\n");
         ASSERT_EQUALS("", errout.str());
 
+        check("void f(std::vector<std::vector<int>>& v) {\n"
+              "    auto it = v.begin();\n"
+              "    it->push_back(1);\n"
+              "}\n");
+        ASSERT_EQUALS("", errout.str());
```

Comment:
I am not sure what FP this is fixing, but maybe it would be better to use `ValueType` instead of `typeDecl`.

Reply:
Currently there is a `checkLibraryFunction` warning for `push_back()`. `typeDecl()` is called by `astCanonicalType()`.